crates/ibc/src/core/ics24_host/identifier.rs

```diff
@@ -518,27 +540,81 @@ impl std::error::Error for IdentifierError {}
 
 #[cfg(test)]
 mod tests {
+    use rstest::rstest;
 
     use super::*;
 
-    #[test]
-    fn test_valid_chain_id() {
-        assert!(ChainId::from_str("chainA-0").is_ok());
-        assert!(ChainId::from_str("chainA-1").is_ok());
-        assert!(ChainId::from_str("chainA--1").is_ok());
-        assert!(ChainId::from_str("chainA-1-2").is_ok());
+    #[rstest]
```

Comment:
Would you add a case to ensure that `ChainId::new_with_revision_format("chainA-2", 0)` fails?